src/Parser.js

```diff
@@ -789,7 +789,7 @@ Parser.prototype.parseSizeGroup = function(optional) {
     let res;
     if (!optional && this.nextToken.text !== "{") {
         res = this.parseRegexGroup(
-            /^[-+]? *(?:$|\d+|\d+\.\d*|\.\d*) *[a-z]{0,2}$/, "size");
+            /^[-+]? *(?:$|\d+|\d+\.\d*|\.\d*) *[a-z]{0,2}\s*$/, "size");
```

Comment:
Ah, but whitespace conversion is already handled by KaTeX's lexer, which changes all `[ \r\n\t]+` sequences to a single space character.

New commit has test making this clear.